phpstan-baseline.neon

```diff
@@ -0,0 +1,532 @@
+			count: 1
+			path: eZ/Publish/Core/MVC/Symfony/Templating/Tests/Twig/Extension/FileSystemTwigIntegrationTestCase.php
+
+		-
+			message: "#^Class eZ\\\\Publish\\\\Core\\\\Persistence\\\\Legacy\\\\Content\\\\FieldValue\\\\Converter\\\\SelectionConverter constructor invoked with 0 parameters, 1 required\\.$#"
+			count: 1
+			path: eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/SelectionConverter.php
+
+		-
+			message: "#^Method eZ\\\\Publish\\\\Core\\\\Persistence\\\\Legacy\\\\Content\\\\Handler\\:\\:deleteContent\\(\\) should return bool but return statement is missing\\.$#"
+			count: 1
+			path: eZ/Publish/Core/Persistence/Legacy/Content/Handler.php
+
+		-
+			message: "#^Method eZ\\\\Publish\\\\Core\\\\Persistence\\\\Legacy\\\\Content\\\\Handler\\:\\:deleteVersion\\(\\) should return bool but return statement is missing\\.$#"
+			count: 1
+			path: eZ/Publish/Core/Persistence/Legacy/Content/Handler.php
+
+		-
+			message: "#^Method eZ\\\\Publish\\\\Core\\\\Persistence\\\\Legacy\\\\Content\\\\Location\\\\Handler\\:\\:move\\(\\) should return bool but return statement is missing\\.$#"
+			count: 1
+			path: eZ/Publish/Core/Persistence/Legacy/Content/Location/Handler.php
+
+		-
+			message: "#^Method eZ\\\\Publish\\\\Core\\\\Persistence\\\\Legacy\\\\Content\\\\Location\\\\Handler\\:\\:removeSubtree\\(\\) should return bool but return statement is missing\\.$#"
+			count: 1
+			path: eZ/Publish/Core/Persistence/Legacy/Content/Location/Handler.php
+
+		-
+			message: "#^Method eZ\\\\Publish\\\\Core\\\\Persistence\\\\Legacy\\\\Content\\\\Location\\\\Handler\\:\\:swap\\(\\) should return bool but return statement is missing\\.$#"
+			count: 1
+			path: eZ/Publish/Core/Persistence/Legacy/Content/Location/Handler.php
+
+		-
+			message: "#^Method eZ\\\\Publish\\\\Core\\\\Persistence\\\\Legacy\\\\Content\\\\TreeHandler\\:\\:removeSubtree\\(\\) should return bool but return statement is missing\\.$#"
+			count: 1
+			path: eZ/Publish/Core/Persistence/Legacy/Content/TreeHandler.php
+
+		-
+			message: "#^Method eZ\\\\Publish\\\\Core\\\\Persistence\\\\Legacy\\\\Content\\\\Type\\\\Handler\\:\\:addFieldDefinition\\(\\) should return eZ\\\\Publish\\\\SPI\\\\Persistence\\\\Content\\\\Type\\\\FieldDefinition but return statement is missing\\.$#"
+			count: 1
+			path: eZ/Publish/Core/Persistence/Legacy/Content/Type/Handler.php
+
+		-
+			message: "#^Undefined variable\\: \\$loader$#"
+			count: 2
+			path: eZ/Publish/Core/Persistence/Legacy/Tests/HandlerTest.php
+
+		-
+			message: "#^Class eZ\\\\Publish\\\\SPI\\\\Persistence\\\\Content\\\\UrlAlias referenced with incorrect case\\: eZ\\\\Publish\\\\SPI\\\\Persistence\\\\Content\\\\URLAlias\\.$#"
+			count: 2
+			path: eZ/Publish/Core/Repository/URLAliasService.php
+
+		-
+			message: "#^Class Symfony\\\\Component\\\\Validator\\\\ConstraintViolation constructor invoked with 1 parameter, 6\\-10 required\\.$#"
+			count: 1
+			path: eZ/Publish/SPI/FieldType/Generic/Tests/GenericTest.php
+
```

Comment:
That baseline is suspiciously short :trollface:

Reply:
It's the lowest level of analysis possible (`0`). Only the most glaring and obvious issues are reported.

Reply:
See [PHPStan rule levels](https://phpstan.org/user-guide/rule-levels):

> Here’s a brief overview of what’s checked on each level. Levels are cumulative - for example running level 5 also gives you all the checks from levels 0-4.
> 
> 0. basic checks, unknown classes, unknown functions, unknown methods called on $this, wrong number of arguments passed to those methods and functions, always undefined variables
> 1. possibly undefined variables, unknown magic methods and properties on classes with __call and __get
> 2. unknown methods checked on all expressions (not just $this), validating PHPDocs
> 3. return types, types assigned to properties
> 4. basic dead code checking - always false instanceof and other type checks, dead else branches, unreachable code after return; etc.
> 5. checking types of arguments passed to methods and functions
> 6. report missing typehints
> 7. report partially wrong union types - if you call a method that only exists on some types in a union type, level 7 starts to report that; other possibly incorrect situations
> 8. report calling methods and accessing properties on nullable types
> 9. be strict about the mixed type - the only allowed operation you can do with it is to pass it to another mixed

Reply:
I am a little anxious already by reading points 1 and 2. 😜 Nice improvement though!

Reply:
FYI
```
composer phpstan -- --level=1
...
[ERROR] Found 69 errors
```
```
composer phpstan -- --level=2
...
[ERROR] Found 1444 errors
```
Which is to be expected: a lot of PHPDoc's do not match their actual contents.